### PR TITLE
feat(dedicated.carbon): add feature flipping for legacy menu

### PR DESCRIPTION
**Pull request by darsene**
ref: MANAGER-11510

| Question         | Answer
| ---------------- | ---
| Branch?          | `feat/carbon-calculator`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix #MANAGER-11510
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- ~Only FR translations have been updated~
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- ~Breaking change is mentioned in relevant commits~

### CODE REVIEW

File: packages/manager/apps/container/src/container/legacy/server-sidebar/universe/AccountSidebar.tsx [hunk 1]
```diff
@@ -24,6 +24,16 @@ export default function AccountSidebar() {
   const region = environment.getRegion();
   const isEnterprise = environment.getUser()?.enterprise;
 
+  const getFeatures = (): Promise<Record<string, string>> =>
+    reketInstance.get(`/feature/${features.join(',')}/availability`, {
+      requestType: 'aapi',
+    });
+
+  const { data: availability } = useQuery(
```
Comment (mohammed-zahaf) [inline packages/manager/apps/container/src/container/legacy/server-sidebar/universe/AccountSidebar.tsx:32]:
```suggestion
  const { data: sidebarDedicateAvailability } = useQuery(
```

File: packages/manager/apps/container/src/container/legacy/server-sidebar/universe/AccountSidebar.tsx [hunk 2]
```diff
@@ -87,12 +97,13 @@ export default function AccountSidebar() {
         routeMatcher: new RegExp('^/contacts'),
       });
 
-      menu.push({
-        id: 'my-carbon-footprint',
-        label: t('sidebar_carbon_footprint'),
-        href: navigation.getURL('carbon-calculator', '/'),
-        routeMatcher: new RegExp('^/carbon-calculator'),
-      });
+      if (availability && availability['carbon-calculator']) {
```
Comment (mohammed-zahaf) [inline packages/manager/apps/container/src/container/legacy/server-sidebar/universe/AccountSidebar.tsx:100]:
```suggestion
      if (availability && sidebarDedicateAvailability['carbon-calculator']) {
```

File: packages/manager/apps/container/src/container/legacy/server-sidebar/universe/AccountSidebar.tsx (hunk 1 above)
Comment (darsene) [reply packages/manager/apps/container/src/container/legacy/server-sidebar/universe/AccountSidebar.tsx]: fixed!

File: packages/manager/apps/container/src/container/legacy/server-sidebar/universe/AccountSidebar.tsx (hunk 2 above)
Comment (darsene) [reply packages/manager/apps/container/src/container/legacy/server-sidebar/universe/AccountSidebar.tsx]: fixed!

**Review by ghyenne — APPROVED** [review]

**Review by mohammed-zahaf — APPROVED** [review]

**Pull request merged**